pyccel/errors/messages.py

```diff
@@ -73,11 +73,12 @@
 PYCCEL_RESTRICTION_TRY_EXCEPT_FINALLY = 'Uncovered try/except/finally statements by Pyccel'
 PYCCEL_RESTRICTION_RAISE = 'Uncovered raise statement by Pyccel'
 PYCCEL_RESTRICTION_YIELD = 'Uncovered yield statement by Pyccel'
-PYCCEL_RESTRICTION_IS_RHS = 'Only booleans and None are allowed as rhs for is statement'
+PYCCEL_RESTRICTION_IS_RHS = 'Only booleans, integers and None are allowed as rhs/lhs for is statement'
 PYCCEL_RESTRICTION_IMPORT = 'Import must be inside a def statement or a module'
 PYCCEL_RESTRICTION_IMPORT_IN_DEF = 'Only From Import is allowed inside a def statement'
 PYCCEL_RESTRICTION_IMPORT_STAR = 'import * not allowed'
 PYCCEL_RESTRICTION_OPTIONAL_NONE = 'Variables cannot be equal to None unless they are optional arguments and None is the default value'
+PYCCEL_RESTRICTION_NONE = 'Cannot compare None to None'
```

Comment:
I don't understand this error.
```
In [1]: None is None                                                                                                               
Out[1]: True

In [2]: None == None                                                                                                               
Out[2]: True
```

Reply:
this error, manage the issue https:#426

Reply:
Although Is(Nil, Nil) is a special case which we need to catch, there is nothing stopping us from returning BooleanTrue() when we encounter it instead of raising an error

Reply:
yes thats true, returning BooleanTrue() is the best way